src/test/java/org/jenkinsci/plugins/gitclient/GitClientTest.java

```diff
@@ -2026,6 +2030,294 @@ public void testFetchOverwritesExistingTags() throws Exception {
+                cliGitAPIImplTest.new VersionTest(false, 1, 7, 1,  1),
+                cliGitAPIImplTest.new VersionTest(false, 1, 7, 2,  0)
+        };
+        cliGitAPIImplTest.doTest("git version 1.7.1", versions);
+    }
+
+    @Test
+    public void test_git_version_opensuse_13() {
+        if (!gitImplName.equals("git")) {
+            return;
+        }
+        CliGitAPIImplTest.VersionTest[] versions = {
+                cliGitAPIImplTest.new VersionTest(true,  1, 8, 4, 5),
+                cliGitAPIImplTest.new VersionTest(true,  1, 8, 4, 4),
+                cliGitAPIImplTest.new VersionTest(false, 1, 8, 4, 6)
+        };
+        cliGitAPIImplTest.doTest("git version 1.8.4.5", versions);
+    }
+
+    @Test
+    public void test_git_version_ubuntu_13() {
+        if (!gitImplName.equals("git")) {
+            return;
+        }
+        CliGitAPIImplTest.VersionTest[] versions = {
+                cliGitAPIImplTest.new VersionTest(true,  1, 8, 3, 2),
+                cliGitAPIImplTest.new VersionTest(true,  1, 8, 3, 1),
+                cliGitAPIImplTest.new VersionTest(false, 1, 8, 3, 3)
+        };
+        cliGitAPIImplTest.doTest("git version 1.8.3.2", versions);
+    }
+
+    @Test
+    public void test_git_version_ubuntu_14_04_ppa() {
+        if (!gitImplName.equals("git")) {
+            return;
+        }
+        CliGitAPIImplTest.VersionTest[] versions = {
+                cliGitAPIImplTest.new VersionTest(true,  2, 2, 2, 0),
+                cliGitAPIImplTest.new VersionTest(true,  2, 2, 1, 0),
+                cliGitAPIImplTest.new VersionTest(false, 2, 2, 3, 0)
+        };
+        cliGitAPIImplTest.doTest("git version 2.2.2", versions);
+    }
+
+    @Test
+    public void test_git_version_ubuntu_14_04_ppa_2_3_0() {
+        if (!gitImplName.equals("git")) {
+            return;
+        }
+        CliGitAPIImplTest.VersionTest[] versions = {
+                cliGitAPIImplTest.new VersionTest(true,  2, 3, 0, 0),
+                cliGitAPIImplTest.new VersionTest(true,  2, 2, 9, 0),
+                cliGitAPIImplTest.new VersionTest(false, 2, 3, 1, 0)
+        };
+        cliGitAPIImplTest.doTest("git version 2.3.0", versions);
+    }
+
+    @Test
+    public void test_git_version_ubuntu_14_04_ppa_2_3_5() {
+        if (!gitImplName.equals("git")) {
+            return;
+        }
+        CliGitAPIImplTest.VersionTest[] versions = {
+                cliGitAPIImplTest.new VersionTest(true,  2, 3, 5, 0),
+                cliGitAPIImplTest.new VersionTest(true,  2, 2, 9, 9),
+                cliGitAPIImplTest.new VersionTest(false, 2, 3, 5, 1),
+                cliGitAPIImplTest.new VersionTest(false, 2, 4, 0, 0),
+                cliGitAPIImplTest.new VersionTest(false, 3, 0, 0, 0)
+        };
+        cliGitAPIImplTest.doTest("git version 2.3.5", versions);
+    }
+
+    @Test
+    public void test_git_ssh_executable_found_on_windows() {
+        if (!gitImplName.equals("git")) {
+            return;
+        }
```

Comment:
JUnit offers a more elegant way to skip a test when some condition is incorrect.

```
        assumeThat(gitImplName, is("git"));
```

It is used elsewhere in the file and is understood by test reporting frameworks better than the early return technique.

Reply:
I just saw AssumeThat, changed it. Thanks!